### PR TITLE
new_pangram.py

### PR DESCRIPTION
**Pull request by amymhaddad**
There are two versions of the pangram exercise in this file.

### CODE REVIEW

File: new_pangram.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    """Is the sentence a pangram?"""
+
+# Option 1: for loop
+    letters_in_alphabet = string.ascii_lowercase
+    normalized_sentence = sentence.lower().replace(' ', '')
+
+    if sentence == '':
+        return False
+
+    for letter in letters_in_alphabet:
```
Comment (amymhaddad) [inline new_pangram.py]: @paulghaddad I was trying to make a list comprehension in place of this for loop and couldn't get it to work with the return statement. Here's what I had: 
missing_letter = [return False for letter in letters_in_alphabet if letter not in normalized_sentence]

Is a list comprehension possible with a "return False"?

File: new_pangram.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        if letter not in normalized_sentence:
+            return False
+    return True
+
+#Option 2: using set
```
Comment (amymhaddad) [inline new_pangram.py]: @paulghaddad I didn't find it advantageous to use a while loop for this second version. I think it complicates the code. Thoughts?

File: new_pangram.py (hunk 1 above)
Comment (paulghaddad) [reply new_pangram.py]: @amymhaddad You'll only want to build up a list comprehension if you're _only_ creating a new object. In this case, you're not doing that. Instead, you're iterating through each of the letters and returning `False` early if needed. So in this case a `for` loop is best.

File: new_pangram.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+
+"""Tests to see if a sentence is a pangram"""
+
+import string
+
+def is_pangram(sentence):
+    """Is the sentence a pangram?"""
+
+# Option 1: for loop
```
Comment (paulghaddad) [inline new_pangram.py]: This option is perfect!

File: new_pangram.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    for letter in letters_in_alphabet:
+        if letter not in normalized_sentence:
+            return False
+    return True
+
+#Option 2: using set
+
+    letters_in_alphabet_set = set(string.ascii_lowercase)
+    normalized_sentence_set = set(sentence.lower().replace(' ', ''))
+    difference = letters_in_alphabet_set & normalized_sentence_set
```
Comment (paulghaddad) [inline new_pangram.py]: So the `&` is the intersection between these two sets, so a better variable name would be `intersection` perhaps.

File: new_pangram.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    if difference == letters_in_alphabet_set:
```
Comment (paulghaddad) [inline new_pangram.py]: Lines 27-29 can be expressed more succinctly. We've seen this pattern before:

```
return difference == letters_in_alphabet_set
```

Recall `difference == letters_in_alphabet_set` returns a Boolean, so you can just return that directly.

**Comment by amymhaddad** [comment]
@paulghaddad  Ready for final review!

**Pull request closed without merging**